crypto/s2n_rsa_pss.c

```diff
@@ -176,14 +176,20 @@ static int s2n_rsa_pss_keys_match(const struct s2n_pkey *pub, const struct s2n_p
 
 static int s2n_rsa_pss_key_free(struct s2n_pkey *pkey)
 {
-    /* This object does not own the reference to the key --
-     * s2n_pkey handles it. */
+    POSIX_ENSURE_REF(pkey);
+    POSIX_ENSURE_REF(&pkey->key.rsa_key);
+    struct s2n_rsa_key *rsa_key = &pkey->key.rsa_key;
```

Comment:
This is a little odd. "&pkey->key.rsa_key" can never be null. You're asking if the known memory location of a field in a struct is null.

If you just want to make it clear to someone skimming that rsa_key is safe to dereference (without them needing to know how it was set), I'd do:
```suggestion
    POSIX_ENSURE_REF(pkey);
    struct s2n_rsa_key *rsa_key = &pkey->key.rsa_key;
    POSIX_ENSURE_REF(rsa_key);
```
The second POSIX_ENSURE_REF is still unnecessary, but at least it's less obviously unnecessary.

Reply:
A bit of back and forth on this. `POSIX_ENSURE_REF(&pkey->key.rsa_key);` just check that the address is valid `&`.

The reason for this is because pkey->key is a union and can point to rsa or ecdsa and its not possible to know if rsa is the one allocated. This is because there is no type field associated with the union on s2n_pkey.

The way forward here is to just check POSIX_ENSURE_REF(pkey) and assume that rsa_key is the one being used since fields on s2n_pkey can only be [accessed via methods which behave based on initialization](https://github.com/aws/s2n-tls/blob/120920817625c51a983cfced439a501673d143d6/crypto/s2n_rsa_pss.c#L215-L225).